### PR TITLE
Add eta quotients for classical modular forms

### PR DESCRIPTION
**Pull request by davidlowryduda**
This commit adds eta quotients to pages on classical modular newform
orbit pages, if they exist.

The commit message follows.

There are a few design decisions here, and I document my reasons for
them now. Firstly, this is a relatively rare property, and so it should
not be a column in the db associated to each modular form orbit --- it
would simply be "None" for most of them. Thus it should either be its
own table in the db, hardcoded in the file, or in a yaml. I have chosen
to put this in a yaml file, similar to how some parts of the LMFDB parse
a yaml file to generate their code segments.

This yaml file needs to be parsed sometime and somewhere. It could
either be parsed in web_newform.py or in main.py. I've chosen to parse
it as late as possible, exactly in the rendering function in main.py.
But this is more open to interpretation.

There remain a few things before fully live.
1. This makes the bottom_knowls for functions expressible as eta
   products obsolete. These should be removed so that the list of
   "interesting forms" isn't filled with them.
2.  This also reuses a knowl, mf.half_integral_weight.eta_quotients,
   which does not seem to have been revisited in a long time. It may
   need to be rewritten.

See also the discussion in #2783.

### CODE REVIEW

**Comment by AndrewVSutherland** [comment]
Reviewing this now.

**Comment by AndrewVSutherland** [comment]
Thanks for doing this!  I only have two cosmetic suggestions:

 - Display quotients as fractions.
 - Also show the corresponding q-product formula.

See https://beta.lmfdb.org/knowledge/show/cmf.80.2.a.b.bottom for an example of both.

File: lmfdb/classical_modular_forms/main.py [hunk 1]
```diff
@@ -322,14 +323,38 @@ def parse_prec(info):
...
         return abort(404, err.args)
+
     info = to_dict(request.args)
     info['display_float'] = display_float
     info['format'] = info.get('format', 'embed')
+
+    _curdir = os.path.dirname(os.path.abspath(__file__))
+    etaquotients = yaml.load(
+            open(os.path.join(_curdir, "eta.yaml")),
+            Loader=yaml.FullLoader)
```
Comment (edgarcosta) [inline lmfdb/classical_modular_forms/main.py:354]: what about making eta quotients a global for that file and you load the YAML file when it gets imported?

Comment (davidlowryduda) [reply lmfdb/classical_modular_forms/main.py:354]: This makes sense. I'll do that.

**Comment by davidlowryduda** [comment]
@AndrewVSutherland I can do that. I'll update this in the next couple of days.

**Pull request merged**